frontend/src/components/common/Terminal.tsx

```diff
@@ -146,39 +267,89 @@ export default function Terminal(props: TerminalProps) {
     [props.open]
   );
 
+  React.useEffect(() => {
+    if (shells.available.length === 0) {
+      setShells({
+        available: getAvailableShells(),
+        currentIdx: 0,
+      });
+    }
+  }, [item]);
+
+  function getAvailableShells() {
+    const selector = item.spec?.nodeSelector || {};
+    const os = selector['kubernetes.io/os'] || selector['beta.kubernetes.io/os'];
+    if (os === 'linux') {
+      return ['sh', 'bash'];
```

Comment:
Why is `sh` first and not `bash`?

Reply:
I think `sh` is more common than `bash`. My reasoning was in order to minimize the chances of failure.
Also, once you connect to `sh` you can run bash...